diff -raudB ./buttons.php.orig ./buttons.php
--- ./buttons.php.orig	Mon Jan 21 22:18:18 2008
+++ ./buttons.php	Mon Mar  3 14:01:29 2008
@@ -55,7 +55,7 @@
           $sb_show_spam_button_folder_allow_override,
           $sb_show_not_spam_button_folder,
           $sb_show_not_spam_button_folder_allow_override,
-          $extra_buttons;
+          $extra_buttons, $confirm_spam_action, $confirm_ham_action;
  
    $passed_ent_id = 0;
    sqgetGlobalVar('passed_ent_id', $passed_ent_id, SQ_FORM);
@@ -163,7 +163,26 @@
       else if (check_sm_version(1, 5, 1))
          $buttons[1]['isSpam'] = array(0 => _($spam_button_text), 1 => 'submit');
       else
-         $ret .= '<input type="submit" name="isSpam" value="' . _($spam_button_text) . "\" />\n";
+        if ( $confirm_spam_action )
+	{
+    	    $ret .= '<input type="submit" onClick="return confirmSubmit()" name="isSpam" value="' . _($spam_button_text) . "\" />\n
+	    <script LANGUAGE=\"JavaScript\">\n
+	    <!--\n
+		function confirmSubmit()\n
+		{\n
+		    var agree=confirm(\"" . _("Are you sure?") ."\");\n
+		    if (agree)\n
+			return true ;\n
+		    else\n
+		    return false ;\n
+		    }\n
+	    // -->\n
+	    </script>\n";		    
+	}
+	else
+	{
+    	    $ret .= '<input type="submit" name="isSpam" value="' . _($spam_button_text) . "\" />\n";	
+	}
    }
 
 
@@ -194,7 +213,25 @@
       else if (check_sm_version(1, 5, 1))
          $buttons[1]['notSpam'] = array(0 => _($not_spam_button_text), 1 => 'submit');
       else
-         $ret .= '<input type="submit" name="notSpam" value="' . _($not_spam_button_text) . "\" />\n";
+        if ( $confirm_ham_action ) {
+    	    $ret .= '<input type="submit" onClick="return confirmSubmit()"  name="notSpam" value="' . _($not_spam_button_text) . "\" />\n
+	    <script LANGUAGE=\"JavaScript\">\n
+	    <!--\n
+		function confirmSubmit()\n
+		{\n
+		    var agree=confirm(\"" . _("Are you sure?") ."\");\n
+		    if (agree)\n
+			return true ;\n
+		    else\n
+		    return false ;\n
+		    }\n
+	    // -->\n
+	    </script>\n";
+	}
+	else	
+	{
+    	    $ret .= '<input type="submit" name="notSpam" value="' . _($not_spam_button_text) . "\" />\n";
+	}
    }
 
 
@@ -247,7 +284,8 @@
           $sb_show_spam_button_folder,
           $sb_show_spam_button_folder_allow_override,
           $sb_show_not_spam_button_folder, $extra_buttons,
-          $sb_show_not_spam_button_folder_allow_override;
+          $sb_show_not_spam_button_folder_allow_override,
+	  $confirm_spam_action,$confirm_ham_action;
 
    spam_buttons_init();
 
@@ -348,39 +387,69 @@
       && (in_array($mailbox, $sb_show_spam_button_folder)                // 2Ba
        || (empty($sb_show_spam_button_folder)                            // 2Bb
         && !in_array($mailbox, $sb_suppress_spam_button_folder)))))
-    && !($sb_report_spam_by_move_to_folder && !empty($passed_ent_id)))   // 3
-      $spam_links[] = array('URL' => sqm_baseuri() 
-         . 'src/read_body.php?isSpam=yslnk&mailbox=' 
-         . urlencode($mailbox) . '&passed_id=' . $passed_id . '&view_as_html='
-         . $view_as_html . '&startMessage=' . $startMessage . '&passed_ent_id=' 
-         . $passed_ent_id . '&account=' . $account, 
-         'Text' => _($spam_button_text));
+    && !($sb_report_spam_by_move_to_folder && !empty($passed_ent_id))) { // 3
+     
+	if ( $confirm_spam_action ) {
+	    $spam_links[] = array('URL' => sqm_baseuri() 
+            . 'src/read_body.php?isSpam=yslnk&mailbox=' 
+            . urlencode($mailbox) . '&passed_id=' . $passed_id . '&view_as_html='
+            . $view_as_html . '&startMessage=' . $startMessage . '&passed_ent_id=' 
+            . $passed_ent_id . '&account=' . $account .'" onclick="return confirm(\''. _("Are you sure?") .'\')', 
+            'Text' => _($spam_button_text));
+	}
+	else
+	{
+	    $spam_links[] = array('URL' => sqm_baseuri() 
+            . 'src/read_body.php?isSpam=yslnk&mailbox=' 
+            . urlencode($mailbox) . '&passed_id=' . $passed_id . '&view_as_html='
+            . $view_as_html . '&startMessage=' . $startMessage . '&passed_ent_id=' 
+            . $passed_ent_id . '&account=' . $account, 
+            'Text' => _($spam_button_text));
+	}
+	 
+    }
+

    // 1) is button turned on in the first place?
    // 2) either A or B below: 
    //     A) if current message is tagged as ham, don't show ham button
    //     B) if not using message tag settings, then:
    //        a) if in (one of) the spam folder(s) ($sb_show_not_spam_button_folder
    //           defines a list of spam folders to show the ham button in),
    //           show ham button
    //        b) otherwise, if not using the spam folder (list), then
    //           if in (one of) the ham folder(s), don't show ham button
    // 3) can't show button if report method is move-to-folder and we're
    //    looking at a message that is an attachment to another
    //
    if ($show_not_spam_button                                                 // 1
     && (current_message_is_tagged(FALSE) === FALSE                           // 2A
      || (current_message_is_tagged(FALSE) === 0                              // 2B
       && (in_array($mailbox, $sb_show_not_spam_button_folder)                // 2Ba
        || (empty($sb_show_not_spam_button_folder)                            // 2Bb
         && !in_array($mailbox, $sb_suppress_not_spam_button_folder)))))
     && !($sb_report_not_spam_by_move_to_folder && !empty($passed_ent_id)))   // 3
-      $spam_links[] = array('URL' => sqm_baseuri() 
-         . 'src/read_body.php?notSpam=yslnk&mailbox='
-         . urlencode($mailbox) . '&passed_id=' . $passed_id . '&view_as_html='
-         . $view_as_html . '&startMessage=' . $startMessage . '&passed_ent_id=' 
-         . $passed_ent_id . '&account=' . $account, 
-         'Text' => _($not_spam_button_text));
+    {
+        if ( $confirm_ham_action )
+	{    
+	    $spam_links[] = array('URL' => sqm_baseuri() 
+            . 'src/read_body.php?notSpam=yslnk&mailbox='
+            . urlencode($mailbox) . '&passed_id=' . $passed_id . '&view_as_html='
+            . $view_as_html . '&startMessage=' . $startMessage . '&passed_ent_id=' 
+            . $passed_ent_id . '&account=' . $account .'" onclick="return confirm(\''. _("Are you sure?") .'\')', 
+            'Text' => _($not_spam_button_text));
+	}
+	else
+	{
+	    $spam_links[] = array('URL' => sqm_baseuri() 
+            . 'src/read_body.php?notSpam=yslnk&mailbox='
+            . urlencode($mailbox) . '&passed_id=' . $passed_id . '&view_as_html='
+            . $view_as_html . '&startMessage=' . $startMessage . '&passed_ent_id=' 
+            . $passed_ent_id . '&account=' . $account, 
+            'Text' => _($not_spam_button_text));
+	    
+	}
+    }
 
    if (check_sm_version(1, 5, 2))
       $links = array_merge($links, $spam_links);
diff -raudB ./config_example.php.orig ./config_example.php
--- ./config_example.php.orig	Mon Jan  7 02:24:31 2008
+++ ./config_example.php	Tue Mar  4 09:54:08 2008
@@ -669,7 +669,8 @@
 $is_spam_subject_prefix = 'SPAM';
 $is_not_spam_subject_prefix = 'HAM';
 
-
+$confirm_spam_action = 1;
+$confirm_ham_action = 0;
 
 // You may also specify overrides for the SMTP server for the 
 // email only (see SquirrelMail's main configuration file or
diff -raudB ./spam_buttons.pot.orig ./spam_buttons.pot
--- ./spam_buttons.pot.orig	Wed Jan 23 16:25:29 2008
+++ ./spam_buttons.pot	Tue Mar  4 09:49:19 2008
@@ -194,3 +194,6 @@
 
 msgid "Senders have been whitelisted"
 msgstr ""
+
+msgid "Are you sure?"
+msgstr ""

