--- ../../functions/mailbox_display.php.orig	2005-03-05 00:01:23.000000000 -0800
+++ ../../functions/mailbox_display.php	2005-03-05 00:01:07.000000000 -0800
@@ -738,6 +738,7 @@
             getMbxList($imapConnection);
             echo getButton('SUBMIT', 'moveButton',_("Move")) . "\n";
             echo getButton('SUBMIT', 'attache',_("Forward")) . "\n";
+            do_hook('mailbox_display_buttons');
 
   echo "      </small></td>\n"
          . html_tag( 'td', '', 'right', '', 'nowrap' );
