--- ../../functions/mailbox_display.php.orig	2008-01-09 00:35:15.000000000 -0800
+++ ../../functions/mailbox_display.php	2008-01-09 00:57:05.000000000 -0800
@@ -1100,6 +1100,8 @@
         }
         $aFormElements['account']  = array($iAccount,'hidden');
     }
+    $ret = do_hook('message_list_controls', $aFormElements);
+    if (is_array($ret[1])) $aFormElements = $ret[1];
 
     /*
      * This is the beginning of the message list table.
